### PR TITLE
add notes for n26742

### PR DESCRIPTION
**Pull request by stickies-v**
cc @fjahr - very optional, but if you have any comments or suggestions please feel free to provide feedback!

### CODE REVIEW

File: _posts/2023-01-11-#26742.md [hunk 1]
```diff
@@ -7,18 +7,59 @@ authors: [fjahr]
 components: ["rpc/rest/zmq"]
 host: stickies-v
 status: upcoming
-commit:
```
Comment (glozow) [inline _posts/2023-01-11-#26742.md:10]: Re: title, I wonder if it's worth stripping "2nd attempt" and/or "http" ? I usually remove stuff if already labeled in components

File: _posts/2023-01-11-#26742.md [hunk 2]
```diff
@@ -7,18 +7,59 @@ authors: [fjahr]
...
-<!-- TODO: Before meeting, add notes and questions
 ## Notes
 
+- First, we'll have a more general look at Bitcoin Core's HTTP server to build an essential understanding of the area we're working on. Then we'll dive deeper into the specifics of this PR.
+
+### HTTP server
+
+- Since [#5677](https://github.com/bitcoin/bitcoin/pull/5677), Bitcoin Core's [HTTP server](https://github.com/bitcoin/bitcoin/blob/adc41cf3b22f8f168e88ce3ad5f27c1130f12beb/src/httpserver.cpp#L138-L139) is based on [libevent2](https://libevent.org/). Libevent is a general purpose event notification library, but is used in Bitcoin Core specifically for HTTP requests (which it supports natively).
+
+- Much (not all) of the libevent interface is hidden behind our own wrappers. For example, [`HTTPRequest`](https://github.com/bitcoin/bitcoin/blob/296e88225096125b08665b97715c5b8ebb1d28ec/src/httpserver.h#L56) wraps `evhttp_request` and [`HTTPEvent`](https://github.com/bitcoin/bitcoin/blob/296e88225096125b08665b97715c5b8ebb1d28ec/src/httpserver.h#L154) wraps [`event_base`](https://libevent.org/doc/structevent__base.html)
```
Comment (glozow) [inline _posts/2023-01-11-#26742.md:21]:
```suggestion
- Much (not all) of the libevent interface is hidden behind wrappers. For example, [`HTTPRequest`](https://github.com/bitcoin/bitcoin/blob/296e88225096125b08665b97715c5b8ebb1d28ec/src/httpserver.h#L56) wraps `evhttp_request` and [`HTTPEvent`](https://github.com/bitcoin/bitcoin/blob/296e88225096125b08665b97715c5b8ebb1d28ec/src/httpserver.h#L154) wraps [`event_base`](https://libevent.org/doc/structevent__base.html)
```

**Comment by stickies-v** [comment]
Thanks @glozow - always appreciate the nits! Incorporated them all. 

I'll merge already but fjahr - if you have any comments, happy to make amendments later on.

**Comment by fjahr** [comment]
@stickies-v Looks good to me as is. 

Just some further ideas if you are open to add something:
- You could ask people to share their opinion on https://github.com/bitcoin/bitcoin/pull/26742/files#r1064707663, thinking about such a question is good for beginners I think
- You could ask people to confirm that the updated test fails on master, also good to practice that for beginners
- You could also add a small question for people to answer what it means if an http connection is "idle" or something along those lines (lifecycle of an http connection maybe). Some people may only have superficial knowledge of http and then it may be hard to fully grasp the context of the issue.

**Comment by stickies-v** [comment]
Thanks for reviewing @fjahr! 

> You could ask people to confirm that the updated test fails on master, also good to practice that for beginners

Good idea, will ask in the chat.

> You could ask people to share their opinion on https://github.com/bitcoin/bitcoin/pull/26742/files#r1064707663, thinking about such a question is good for beginners I think

Yup, that's what I'm going for with Q4: "What’s the purpose of g_requests? Can you think of any alternative approaches that would achieve the same goal?" - will link to that during the discussion

> You could also add a small question for people to answer what it means if an http connection is "idle" or something along those lines (lifecycle of an http connection maybe). Some people may only have superficial knowledge of http and then it may be hard to fully grasp the context of the issue.

Good idea, will ask in the chat.

**Pull request merged**